src/main/ast-to-doc.js

```diff
@@ -42,6 +42,11 @@ function printAstToDoc(ast, options, alignmentSize = 0) {
   function printGenerically(path, args) {
     const node = path.getValue();
 
+    // When `printGenerically` is used as callback of `AstPath.map()`, it will get `index` as `args`
+    if (printGenerically.isAstPathMapCallback) {
+      args = undefined;
```

Comment:
`printGenerically` is recursive. To me, this looks like `args` will be discarded for all nested `printGenerically` calls inside a `path.map` call. I'm surprised tests didn't fail.

Reply:
Because I restore the old `isAstPathMapCallback` value back after `.map` call, not set to `false`. So it will only set to `true` when calling `.map`, the only way to break it is to call `path.map(printGenerically)` in `printGenerically` function self.

Reply:
E.g. format this code with and without this change:
```
bar =
  (foo,
  (module.exports =
    (generate, generateRandomOptions) =>
    ({ code, sourceType, reproductionData = undefined }) => {}));
```

Reply:
Or this
```
`${function* mySagas() {
  yield effects.takeEvery(rexpress.actionTypes.REQUEST_START, function* ({
    id
  }) {
    console.log("pong");
  });
}}
```

Reply:
You are right. It can be recursive, this is wrong.